udpTester.py

```diff
@@ -358,47 +384,47 @@ def receiver(args):
     print(
         f"Received {totalMsgs} complete messages out of {expectedCount}, lost {lost:.1f}%"
     )
+    print(f"Received {duplicatePackets} duplicate packets")
 
 
 def create_parser():
     # shared args
     usageString = """
-    Running the transmitter:
-        udpTester.py transmitter address outgoing [-h] [-p PORT] [-m MESSAGESIZE] [-t TOTALCOUNT] [-s PACKETSIZE] [-i INTERVAL] [-l LOSSINESS]
+    udpTester.py -transmitter -a ADDRESS -o OUTGOING [-h] [-p PORT] [-t TOTALCOUNT] [-m MESSAGESIZE] [-s PACKETSIZE] [-i INTERVAL] [-l LOSSINESS]
 
-    Running the receiver:
-        udpTester.py receiver address [-h] [-p PORT] [-m MESSAGESIZE] [-t TOTALCOUNT] [-s PACKETSIZE] [-r REPORTINTERVAL] [-b RECEIVEBUFFER] [-q QUIET]
+    udpTester.py -receiver -a ADDRESS [-h] [-p PORT] [-t TOTALCOUNT] [-m MESSAGESIZE] [-s PACKETSIZE] [-b RECEIVEBUFFER] [-r REPORTINTERVAL] [-q QUIET]
 
-    Note that outgoing is a mandatory argument for the transmitter. It is the network interface address used for outgoing packets.
+    Example:
+        python3 udpTester.py -transmitter -a 239.0.0.1 -o 192.168.2.8 -t 200 -m 450 -s 150
+        python3 udpTester.py -receiver -a 239.0.0.1 -t 200 -m 450 -s 150 -b 200000
     """
     parser = argparse.ArgumentParser(
         usage=usageString, description="Test multicast traffic"
```

Comment:
By specifying the 'usage' parameter here you are suppressing the help data argparse so helpfully generates for you. You could specify the `epilog` parameter instead to display your instructions after the built-in help. For example, we now lose the information that `-a ADRESS` is mandatory and that I could also write `--address`.

Secondly, I dislike the -transmitter/-receiver format. They break the `*NIX` convention of single char arguments with a single `-` and words with dual dashes `--`. You can add them as subparsers instead: `https://docs.python.org/3/library/argparse.html#argparse.ArgumentParser.add_subparsers` so you could directly parse the first argument as the mode. This would also allow you to nicely split the one-mode only parameters out to the subparsers, meaning that `python udpTester.py receiver --help` would only print the help of the parameters relevant to the receiver.

Reply:
Yes we can use --transmitter or --receiver here